`grid` fixes

t-bltg opened the pull request:
Some preliminary fixes.

t-bltg commented on test/07_test_panel.jl:
```diff
@@ -34,60 +34,55 @@ end
-                displaysize(stdout)[2] - 1,
-                nothing,
-            )
+        # ----------------------------- text only content ---------------------------- #
```
No, this is because you don't `justify` in these, so this is redundant to put them in the inner loop. As a side note, I find the tests hard to read because of the duplication of the test code. I think it is worth some simplifications.

t-bltg replied:
I've added `Base.size(m::Measure)` returning `(m.w, m.h)` for that purpose.

t-bltg commented:
I've added a [stable random number generator](https://github.com/JuliaRandom/StableRNGs.jl) for the test to be reproducible (`.txt` generated files should be identical when running the tests twice) .
Testing was broken because of the spurious `Manifest.toml` files lying all around, this is fixed.

t-bltg commented:
- don't use `@test something == true`, it is redundant
- don't use `@test something == false`, use `@test !something`
- don't use `@test typeof(x) == y` use `@test x isa y`

FedeClaudi commented on test/08_test_layout.jl:
```diff
@@ -283,55 +270,82 @@ end
+
+    w, h = 20, 10
+    panels = collect(
+        Panel("{on_$c} {/on_$c}", style = "bold yellow", width = w, height = h) for
```
I've added a `background` option to `Panel` if you want to have the panel border and content colored. `Panel(; background="on_red")` is the usage, it's not documented yet but it works.

t-bltg replied:
Mmmh, I can't get it to work.

FedeClaudi commented:
> I think test/07_test_panel.jl needs cleanup of un-necessary comments, and simplifications.

`Panel` is perhaps the most complex thing happening in Term because how how many combinations of contents, sizes, styles, justification... so I've been trying to write tests covering as many of these as possible while still being relatively compact. I agree that it's a bit of a mess though, any pointers on what we could do to improve it?

Thanks for the other changes it all looks great.

t-bltg commented:
I've simplified `07_test_panel.jl` to reduce duplicates, and added more source code simplifications (mostly `if ... end` simplifications, one-liners, short-circuit if, ...) to adhere closer to the `julia` language (blaming `Python` :laughing:).

FedeClaudi commented:
awesome, I had been looking for a way to do assignment with `&&` expressions: `style && (txt = apply_style(txt))` is great to know!

I guess you can tell I am still transitioning to Julia hey :)

t-bltg commented:
> This looks ready, let me know when you're happy for me to merge.

I'll stop now, you can merge before the PR gets too big too handle ...
